spec.html

```diff
@@ -9009,8 +9029,10 @@ <h1>ArraySetLength ( _A_, _Desc_ )</h1>
     <emu-clause id="sec-string-exotic-objects">
       <h1>String Exotic Objects</h1>
       <p>A <em>String object</em> is an exotic object that encapsulates a String value and exposes virtual integer-indexed data properties corresponding to the individual code unit elements of the String value. String exotic objects always have a data property named *"length"* whose value is the number of code unit elements in the encapsulated String value. Both the code unit data properties and the *"length"* property are non-writable and non-configurable.</p>
+
+      <p>An object is a <dfn id="string-exotic-object">String exotic object</dfn> if its [[GetOwnProperty]], [[DefineOwnProperty]], and [[OwnPropertyKeys]] internal methods use the following implementations, and its other internal methods use the implementations found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed in StringCreate.</p>
```

Comment:
I think similar patterns occur in other places. For example, [Integer-indexed exotic object][0], [TypedArray object][1], and [TypedArray instance][2] are terms used/seemingly defined in various places which all describe the same thing\*. Arguably they’re different concepts that just _happen_ to all occur exclusively together, but it’s not always clear to me how it’s decided when one or another term should be used.

> \* Except that the third definition, like in all other ‘instance’ sections, mentions inheriting from particular prototypes. IMO those inheritance sentences are incorrect, since spec algorithms that are specific to instances of a particular kind never rely on the result of [[GetPrototypeOf]]. In other words, statements like “Each element of a TypedArray instance has the same underlying binary scalar data type” are, in a strict reading, inapplicable unless the object in question inherits from a TypedArray prototype, but that is surely not the intended meaning.

[0]: https://tc39.es/ecma262/#integer-indexed-exotic-object
[1]: https://tc39.es/ecma262/#sec-typedarray-objects
[2]: https://tc39.es/ecma262/#sec-properties-of-typedarray-instances

Reply:
Indeed, but that situation predates this PR, so it might be worth raising as a separate issue.

Reply:
> Should we delete the <em> tags in the previous para? It might also be good to add "(or String object)" to this para.

This sounds good to me.

> (Ultimately, it would probably be good to settle on a single term, but it doesn't have to be this PR's job to accomplish that.)

I agree that we should settle on a single noun ("object"), but I am also fine with e.g. "Array object" being interchangeable with "Array exotic object" given that there are no Array ordinary objects. The "Array object" phrase should link to the correct definition, however.

Mind filing an issue for this?

Reply:
> Mind filing an issue for this?

Done: issue #1818